server/src/main/java/io/crate/expression/scalar/VectorSimilarityFunction.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.data.Input;
+import io.crate.metadata.Functions;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+import io.crate.types.FloatVectorType;
+
+public class VectorSimilarityFunction extends Scalar<Float, float[]> {
+
+    public static void register(Functions.Builder module) {
+        module.add(
+            Signature.scalar(
+                "vector_similarity",
+                FloatVectorType.INSTANCE_ONE.getTypeSignature(),
+                FloatVectorType.INSTANCE_ONE.getTypeSignature(),
+                DataTypes.FLOAT.getTypeSignature()
+            ).withFeature(Scalar.Feature.NULLABLE),
+            VectorSimilarityFunction::new
```

Comment:
Can't we use `BinaryScalar` instead?

Reply:
No, because Input/Return types are different.